src/lib/JreSetupManager.ts

```diff
@@ -24,16 +26,29 @@ class Config extends ConfigFile<ConfigFile.Options> {
     }
 }
 
+// Exported only to be used by tests
+export class JreSetupManagerDependencies {
+    autoDetectJavaHome(handleDetectedValue: (err: Error, home: string) => string): Promise<string> {
```

Comment:
Can't this be simplified? 
```suggestion
    autoDetectJavaHome(): Promise<string> {
    return new Promise<string>((resolve) => {
            findJavaHome({allowJre: true}, (err, home) => {
                resolve(err ? null : home);
            });
        });
    }
```

Reply:
Hmm, that's the portion of logic I attempted to unit test. Your sample code does look cleaner. Let me change my code.
Hopefully, this part of code will not need additional logic or changes.

Reply:
Fixed